transport/conn.go

```diff
@@ -202,6 +234,61 @@ func (c *Conn) Startup(options frame.StartupOptions) (frame.Response, error) {
 	return c.sendRequest(&Startup{Options: options}, false, false)
 }
 
+type QueryResult struct {
+	Rows        []frame.Row
+	Warnings    []string
+	TracingID   frame.UUID
+	PagingState frame.Bytes
+	ColSpec     []frame.ColumnSpec
+}
+
+type statement struct {
+	Content           string
+	Values            []frame.Value
+	PageSize          frame.Int
+	Consistency       frame.Consistency
+	SerialConsistency frame.Consistency
+	Tracing           bool
+	Compression       bool
+}
+
+func (c *Conn) Query(s statement, pagingState frame.Bytes) (QueryResult, error) {
+	req := &Query{
+		Query:       s.Content,
+		Consistency: s.Consistency,
+		Options: frame.QueryOptions{
+			Values:            s.Values,
+			SerialConsistency: s.SerialConsistency,
+			PagingState:       pagingState,
+			PageSize:          s.PageSize,
+		},
+	}
+
+	res, err := c.sendRequest(req, s.Compression, s.Tracing)
+	if err != nil {
+		return QueryResult{}, err
+	}
+
+	return intoQueryResult(res)
+}
+
+func intoQueryResult(res frame.Response) (QueryResult, error) {
```

Comment:
This should be called makeQueryResult.
I'd suggest to create a query.go and put all query related structs and helper functions / constructors there.